<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpenGLVisualizerClass</class>
 <widget class="QMainWindow" name="OpenGLVisualizerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>626</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>OpenGLVisualizer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="1" column="3">
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QCheckBox" name="checkBoxLighting">
          <property name="text">
           <string>Enable Lighting</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QRadioButton" name="radioButtonCW">
          <property name="text">
           <string>CW</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">frontFace</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButtonCCW">
          <property name="text">
           <string>CCW</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">frontFace</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="labelNear">
          <property name="text">
           <string>Near Plane</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditNear">
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>0.1</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="labelFar">
          <property name="text">
           <string>Far Plane</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditFar">
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>10000</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QLabel" name="labelFOVX">
          <property name="text">
           <string>FOV X</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="sliderFOVX">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>179</number>
          </property>
          <property name="value">
           <number>45</number>
          </property>
          <property name="tracking">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::NoTicks</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QLabel" name="labelFOVY">
          <property name="text">
           <string>FOV Y</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="sliderFOVY">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>179</number>
          </property>
          <property name="value">
           <number>45</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="labelRotX">
            <property name="text">
             <string>Rotate U</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="sliderRotX">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="minimum">
             <number>-45</number>
            </property>
            <property name="maximum">
             <number>45</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="pageStep">
             <number>15</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksAbove</enum>
            </property>
            <property name="tickInterval">
             <number>15</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelTransX">
            <property name="text">
             <string>Translate U</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="sliderTransX">
            <property name="minimum">
             <number>-500</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::NoTicks</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="labelRotY">
            <property name="text">
             <string>Rotate V</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="sliderRotY">
            <property name="minimum">
             <number>-45</number>
            </property>
            <property name="maximum">
             <number>45</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="pageStep">
             <number>15</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksAbove</enum>
            </property>
            <property name="tickInterval">
             <number>15</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelTransY">
            <property name="text">
             <string>Translate V</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="sliderTransY">
            <property name="minimum">
             <number>-500</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="labelRotZ">
            <property name="text">
             <string>Rotate N</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="sliderRotZ">
            <property name="minimum">
             <number>-45</number>
            </property>
            <property name="maximum">
             <number>45</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="pageStep">
             <number>16</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksAbove</enum>
            </property>
            <property name="tickInterval">
             <number>15</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelTransZ">
            <property name="text">
             <string>Translate N</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="sliderTransZ">
            <property name="minimum">
             <number>-500</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBoxFixedCenter">
        <property name="text">
         <string>Fixed Center</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QRadioButton" name="radioButtonFill">
          <property name="text">
           <string>Polygons</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">polygonMode</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButtonLine">
          <property name="text">
           <string>Wireframe</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">polygonMode</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButtonPoint">
          <property name="text">
           <string>Points</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">polygonMode</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <widget class="OpenGLWindowWidget" name="openGLWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>2</horstretch>
        <verstretch>2</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="0" column="3">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QSlider" name="sliderRed">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="value">
             <number>255</number>
            </property>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::NoTicks</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelRed">
            <property name="text">
             <string> R</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QSlider" name="sliderGreen">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelGreen">
            <property name="text">
             <string>G</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QSlider" name="sliderBlue">
            <property name="maximum">
             <number>255</number>
            </property>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelBlue">
            <property name="text">
             <string>B</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="buttonLoadFile">
        <property name="text">
         <string>Load Model</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonResetCamera">
        <property name="text">
         <string>Reset Camera</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLWindowWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>OpenGLWindowWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="openglvisualizer.qrc"/>
 </resources>
 <connections/>
 <buttongroups>
  <buttongroup name="polygonMode"/>
  <buttongroup name="frontFace"/>
 </buttongroups>
</ui>
